src/rpc/client.cpp

```diff
@@ -225,11 +225,14 @@ class CRPCConvertTable
 public:
     CRPCConvertTable();
 
-    bool convert(const std::string& method, int idx) {
-        return (members.count(std::make_pair(method, idx)) > 0);
+    /* Return value as UniValue, and first parse it if it is a non-string parameter */
+    UniValue ParseIfNonString(const std::string& value, const std::string& method, int idx) {
```

Comment:
In commit "refactor: use convenience fn to auto parse non-string parameters" (32bfd59c28c61dab46fc204145aa52f8db2b7860)

The name `ParseIfNonString` sounds a little nonsensical to me because it reads like "parse value if value is not a string", but it really means "parse value if parameter type is non-string". Probably hard to choose a better name, but I might go with something like "ParseJsonOrStringParam" or "MaybeParseJson" or "ArgToUniValue" if anything like that sounds better.

Reply:
I agree there's room for misunderstanding here. I think `ParseJsonOrStringParam` doesn't help with clarity, and in general I don't like `Maybe` functions if alternatively the condition(s) can realistically be described in the function name (which I think is possible here). I would be okay with `ArgToUniValue` - or what about `ParseIfNonStringParam`?

Reply:
Renamed to `ParseIfNonStringParam()` - happy to further discuss a better name though. Marking as resolved until then.